validator_instance.go

```diff
@@ -409,7 +409,10 @@ func (v *Validate) StructPartialCtx(ctx context.Context, s interface{}, fields .
 		if len(flds) > 0 {
 
 			vd.misc = append(vd.misc[0:0], name...)
-			vd.misc = append(vd.misc, '.')
+			// Don't append empty name for composed structs
+			if string(vd.misc) != "" {
```

Comment:
Can we check the length of vd.misc here instead of casting to a string? It would be one less allocation :)